Backpack-SwiftUI/Calendar/Classes/Core/CalendarSelectionHandler.swift

```diff
@@ -0,0 +1,35 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+public protocol CalendarSelectionHandler {
```

Comment:
I think this is an unnecesary abstraction. It should be 2 separate properties for the user to pass, not be forced to give 2, when it may not be needed.
I appreciate why you introduced a protocol, but for this case I think is not recommended.

Reply:
If we expand the number of date selection types (like multi selection), it would be harder to handle all those properties at the end

Reply:
That'd not be single nor multiple selection though right?
that's the same problem, if i have a single selection calendar, then it'd force me to implement 3 handlers where i only need one.